ferrocene/tools/self-test/src/binaries.rs

```diff
@@ -25,15 +25,18 @@ fn check_binary(
     name: &str,
     hash: CommitHashOf,
 ) -> Result<(), Error> {
+    /// File permission the binary should have. The symbolic value is `r-xr-xr-x`.`
```

Comment:
not exactly correct, since these are just the minimum required permissions... `0o777` would also work

Reply:
To be honest I do not really understand how the check works.

```rust
metadata.permissions().mode() & MODE != MODE
```

Can you explain?

Reply:
the check ensures that at least the file is both readable and executable...
```
0o5 = 0b101 (r-x)
         ^ we do not care whether or not this, the write bit, is set
0b101 & 0b111 = 0b101
0b101 & 0b101 = 0b101
0b101 & 0b110 = 0b100 (fail, not executable)
0b101 & 0b011 = 0b001 (fail, not readable)

Reply:
Thank you! (I also changed the doc-comment).